lib/commons/color/get-background-color.js

```diff
@@ -176,236 +284,74 @@ function includeMissingElements(elmStack, elm) {
-		return [boundingStack];
-	}
-
-	// Handle inline elements spanning multiple lines to be evaluated
-	let filteredArr = rects
-		.filter(rect => {
-			// exclude manual line breaks in Chrome/Safari
-			return rect.width && rect.width > 0;
-		})
-		.map(rect => {
-			let coords = color.getCoords(rect);
-			if (coords) {
-				return dom.shadowElementsFromPoint(coords.x, coords.y);
-			}
-		});
-
-	if (filteredArr.some(stack => stack === undefined)) {
-		// Can be happen when one or more of the rects sits outside the viewport
-		return null;
-	}
+function calculateObscuringAlpha(elmIndex, elmStack, originalElm) {
+	var totalAlpha = 0;
 
-	// add bounding client rect stack for comparison later
-	filteredArr.splice(0, 0, boundingStack);
-	return filteredArr;
-};
-/**
- * Get filtered stack of block and inline elements, excluding line breaks
- * @method filteredRectStack
- * @memberof axe.commons.color
- * @instance
- * @param {Element} elm
- * @return {Array}
- */
-color.filteredRectStack = function(elm) {
-	let rectStack = color.getRectStack(elm);
-	if (rectStack && rectStack.length === 1) {
-		// default case, elm.getBoundingClientRect()
-		return rectStack[0];
-	} else if (rectStack && rectStack.length > 1) {
-		let boundingStack = rectStack.shift();
-		let isSame;
-		// iterating over arrays of DOMRects
-		rectStack.forEach((rectList, index) => {
-			if (index === 0) {
-				return;
+	if (elmIndex > 0) {
+		// there are elements above our element, check if they contribute to the background
+		for (var i = elmIndex - 1; i >= 0; i--) {
+			let bgElm = elmStack[i];
+			let bgElmStyle = window.getComputedStyle(bgElm);
+			let bgColor = color.getNonAlphaBlendedBackgroundColor(bgElm, bgElmStyle);
+			if (bgColor.alpha && contentOverlapping(originalElm, bgElm)) {
+				totalAlpha += bgColor.alpha;
+			} else {
+				// remove elements not contributing to the background
+				elmStack.splice(i, 1);
 			}
-			// if the stacks are the same, use the first one. otherwise, return null.
-			let rectA = rectStack[index - 1],
-				rectB = rectStack[index];
-
-			// if elements in clientRects are the same
-			// or the boundingClientRect contains the differing element, pass it
-			isSame =
-				rectA.every(function(element, elementIndex) {
-					return element === rectB[elementIndex];
-				}) || boundingStack.includes(elm);
-		});
-		if (!isSame) {
-			axe.commons.color.incompleteData.set('bgColor', 'elmPartiallyObscuring');
-			return null;
 		}
-		// pass the first stack if it wasn't partially covered
-		return rectStack[0];
-	} else {
-		// rect outside of viewport
-		axe.commons.color.incompleteData.set('bgColor', 'outsideViewport');
-		return null;
 	}
-};
-/**
- * Get all elements rendered underneath the current element, In the order they are displayed (front to back)
- * @method getBackgroundStack
- * @memberof axe.commons.color
- * @instance
- * @param {Element} elm
- * @return {Array}
- */
-color.getBackgroundStack = function(elm) {
```

Comment:
Removing this from `color` is a breaking change. I get what you're going for, but let's not go there.